packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -366,7 +379,19 @@ class RenderEditable extends RenderBox {
 
   @override
   double computeMaxIntrinsicHeight(double width) {
-    return _preferredLineHeight * maxLines;
+    if (maxLines != null)
+      return _preferredLineHeight * maxLines;
+    if (width == double.INFINITY) {
+      final String text = _textPainter.text.toPlainText();
+      int lines = 1;
+      for (int index = 0; index < text.length; index += 1) {
+        if (text.codeUnitAt(index) == 0x0A) // count explicit line breaks
```

Comment:
Actually that wouldn't really work. TextPainter's API consistently requires that you lay it out before you do anything like this, and there's nothing in the API that implies a layout. The whole point of this function is that it tries to avoid doing a layout if possible.